src/gpu/sycl/layer_normalizations_kernels.hpp

```diff
@@ -0,0 +1,558 @@
+                float sc = load_float_value(
+                        data_md().data_type(), data_ptr(), s_off);
+                float m = sc - v_mean;
+                v_variance += m * m;
+            }
+            v_variance /= C;
+        }
+
+        float sqrt_variance = sqrtf(v_variance + eps);
+
+        if (idx < conf_.N) {
+            for (dim_t c = 0; c < C; ++c) {
+
+                const float sm = ((conf_.use_scale) ? load_float_value(
+                                          data_scaleshift_md().data_type(),
+                                          scale_ptr(), c)
+                                                    : 1.f)
+                        / sqrt_variance;
+                const float sv = conf_.use_shift
+                        ? load_float_value(data_scaleshift_md().data_type(),
+                                shift_ptr(), c)
+                        : 0;
+                dim_t index = idx * C + c;
+                const auto src_off = data_md().off_l(index);
+                const auto d_off = dst_md().off_l(index);
+                float s = load_float_value(
+                        data_md().data_type(), data_ptr(), src_off);
+                float d = sm * (s - v_mean) + sv;
+
+                float sr = conf_.src_def
+                        ? 1.f
+                        : load_float_value(data_scaleshift_md().data_type(),
+                                rt_oscale_ptr(), 0);
+                float ds = conf_.dst_def
+                        ? 1.f
+                        : load_float_value(data_scaleshift_md().data_type(),
+                                dst_oscale_ptr(), 0);
+                d = (d * sr * (1.f / ds));
+
+                store_float_value(dst_md().data_type(), d, dst_ptr(), d_off);
+            }
+        }
+
+        if (conf_.calculate_stats && conf_.save_stats) {
+            store_float_value(
+                    stat_md().data_type(), v_mean, stat_out_ptr(), s_off);
+            store_float_value(
+                    var_md().data_type(), v_variance, var_out_ptr(), s_off);
+        }
+    }
+
+    sycl_layer_normalization_conf_t conf_;
+    sycl_in_memory_arg_t data_;
+    sycl_in_memory_arg_t scale_;
+    sycl_in_memory_arg_t shift_;
+    sycl_out_memory_arg_t dst_;
+    sycl_out_memory_arg_t mean_out_;
+    sycl_out_memory_arg_t var_out_;
+    sycl_in_memory_arg_t rt_scale_;
+    sycl_in_memory_arg_t dst_scale_;
+};
+
+//backward
+
+struct layer_normalization_bwd_kernel_vec_t {
+    static constexpr int vec_len = 8;
+
+    layer_normalization_bwd_kernel_vec_t(
+            const sycl_layer_normalization_conf_t &conf,
+            sycl_in_memory_arg_t &data, sycl_out_memory_arg_t &diff_data,
+            sycl_in_memory_arg_t &scale, sycl_out_memory_arg_t &diff_scale,
+            sycl_out_memory_arg_t &diff_shift, sycl_in_memory_arg_t &stat,
+            sycl_in_memory_arg_t &var, sycl_in_memory_arg_t &diff_dst,
+            sycl_in_memory_arg_t &dst)
```

Comment:
dst doesn't exist for backward, please remove md and memory for dst in backward pass.